### ISSUE TITLE
[scipy] Add scipy.special.comb to JAX

### ISSUE THREAD

**Issue by Phionx** [google/jax#9709]
Hi!

I've been looking for an implementation of `scipy.special.comb` in JAX, but I couldn't find one. 

If one already exists, please feel free to comment below and close this issue.

Otherwise, here's a draft of a potential implementation:

```
from jax import lax
from jax._src.scipy.special import gammaln
from jax._src.numpy.lax_numpy import _constant_like

def comb(k,n):
    one = _constant_like(k, 1)
    return lax.exp(lax.sub(gammaln(lax.add(k,one)),lax.add(gammaln(lax.add(n,one)), gammaln(lax.add(lax.sub(k,n),one)))))
```

**Comment by jakevdp** [comment]
Hi - this looks like it would be a reasonable addition to `jax.scipy.special`, although we'd want to look into replicating the `exact` and `repetition` arguments as well. Is this something you're interested in contributing via a Pull Request?

**Comment by Phionx** [comment]
Yeah, I'd be happy to put together a PR with the `exact` and `repetition` argument functionality included.

**Comment by PaulScemama** [comment]
@jakevdp @Phionx is this available to work on? Or has it been implemented.

**Comment by Phionx** [comment]
Thanks for the bump @PaulScemama ! 

I had forgotten about this issue and just put together a PR: #18389 to resolve it.
